ui/src/plugin/PluginDetailView.tsx

```diff
@@ -0,0 +1,259 @@
+                        />
+                    </PanelWrapper>
+                ) : null}
+            </DefaultPage>
+        ) : null;
+    }
+}
+
+class PanelWrapper extends Component<{
+    name: string;
+    description?: string;
+    refresh?: () => Promise<void>;
+    icon?: React.ComponentType;
+}> {
+    public render() {
+        return (
+            <Container style={{display: 'block', width: '100%', margin: '20px', color: 'white'}}>
+                <h1>
+                    {this.props.icon ? (
+                        <span>
+                            <this.props.icon />
+                            &nbsp;
+                        </span>
+                    ) : null}
+                    {this.props.name}
+                    {this.props.refresh ? (
+                        <Button
+                            style={{float: 'right'}}
+                            onClick={() => {
+                                this.props.refresh!();
+                            }}>
+                            <Refresh />
+                        </Button>
+                    ) : null}
+                </h1>
+                {this.props.description ? (
+                    <h4 style={{color: 'grey'}}>{this.props.description}</h4>
+                ) : null}
+                <hr />
+                <div
+                    className={this.props.name
+                        .toLowerCase()
+                        .trim()
+                        .replace(/ /g, '-')}>
+                    {this.props.children}
+                </div>
+            </Container>
+        );
+    }
+}
+
+class ConfigurerPanel extends Component<
+    {pluginInfo: IPlugin; initialConfig: string; save: (newConfig: string) => Promise<void>},
+    {unsavedChanges: string}
+> {
+    public state = {unsavedChanges: ''};
+
+    public render() {
+        return (
+            <div>
+                <MonacoEditor
+                    language="yaml"
+                    theme="vs-dark"
+                    height="15em"
+                    ref="monaco"
+                    value={
+                        this.state.unsavedChanges
+                            ? this.state.unsavedChanges
+                            : this.props.initialConfig
+                    }
+                    onChange={(newConfig) => {
+                        this.setState({unsavedChanges: newConfig});
+                    }}
+                />
+                <br />
+                <Button
+                    variant="contained"
+                    color="primary"
+                    fullWidth={true}
+                    disabled={
+                        !this.state.unsavedChanges ||
+                        this.state.unsavedChanges === this.props.initialConfig
+                    }
+                    className="config-save"
+                    onClick={() => {
+                        let newConfig = ((this.refs.monaco as any).editor as any) // tslint:disable-line
```

Comment:
Can't you use the state variable `unsavedChanges` here? Instead of getting the value from the monaco ref?